### PR TITLE
Time series add SST

### PR DESCRIPTION
**Pull request by avalmas-programize**
https://github.com/aqualinkorg/aqualink-app/issues/493

I wasn't able to test the implementation on the sst-backfill script.

### CODE REVIEW

File: packages/api/src/reefs/sources.entity.ts [hunk 1]
```diff
@@ -16,6 +16,7 @@ export enum SourceType {
 
 @Entity()
 @Unique('no_duplicate_sources', ['reef', 'poi', 'type'])
+@Unique('no_duplicate_reef_sources', ['reef', 'type'])
```
Comment (ericboucher) [inline packages/api/src/reefs/sources.entity.ts:19]: What if we have a spotter in two pois?

**Comment by ericboucher** [comment]
Same as spotter, let's make sure we do that more regularly. The goal is to have "almost live" data in our DB

File: packages/api/src/reefs/sources.entity.ts (hunk 1 above)
Comment (avalmas-programize) [reply packages/api/src/reefs/sources.entity.ts]: Yes we needed a unique partial index instead. Fixed.

File: packages/api/package.json [hunk 2]
```diff
@@ -46,7 +46,8 @@
     "backfill-data": "ts-node -r dotenv/config scripts/backfill-daily-data.ts",
```
Comment (ericboucher) [inline packages/api/package.json:46]: let's rename this to "backfill-daily-data" maybe?

File: packages/api/scripts/backfill-sst-sofar.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
+import { ConnectionOptions, createConnection } from 'typeorm';
```
Comment (ericboucher) [inline packages/api/scripts/backfill-sst-sofar.ts:1]: We could merge this script with the spotter data one and add a "task" argument

File: packages/api/scripts/sst-backfill.ts [hunk 4]
```diff
@@ -1,20 +1,29 @@
...
+  insertSSTToTimeSeries,
+} from '../src/utils/time-series.utils';
+import { TimeSeries } from '../src/time-series/time-series.entity';
 
 const dbConfig = require('../ormconfig');
 
 // Reefs and years to backfill SST for
 const yearsArray = [2017, 2018, 2019, 2020];
-const reefsToProcess: number[] = [];
+const reefsToProcess: number[] = [533];
```
Comment (ericboucher) [inline packages/api/scripts/sst-backfill.ts:19]: 👣

File: packages/api/package.json (hunk 2 above)
Comment (avalmas-programize) [reply packages/api/package.json:46]: Done

File: packages/api/scripts/sst-backfill.ts (hunk 4 above)
Comment (avalmas-programize) [reply packages/api/scripts/sst-backfill.ts:19]: Done

File: packages/api/scripts/backfill-sst-sofar.ts (hunk 3 above)
Comment (avalmas-programize) [reply packages/api/scripts/backfill-sst-sofar.ts]: Done
Created new script

File: packages/api/cloud-functions/index.ts [hunk 5]
```diff
@@ -113,3 +114,27 @@ exports.scheduledSpotterTimeSeriesUpdate = functions
       conn.close();
     }
   });
+
+exports.scheduledSSTTimeSeriesUpdate = functions
+  .runWith({ timeoutSeconds: 540 })
+  // Run spotter data update every hour
```
[the remark this answers is not in the document]
Comment (avalmas-programize) [reply packages/api/cloud-functions/index.ts:120]: Done

File: packages/api/scripts/backfill-sofar-time-series.ts [hunk 6]
```diff
@@ -32,7 +66,13 @@ async function run() {
   const sourceRepository = connection.getRepository(Sources);
   const timeSeriesRepository = connection.getRepository(TimeSeries);
 
-  addSpotterData(parsedReefIds, days, connection, {
+  const fn = getTaskFn(task);
+
+  if (!fn) {
+    return null;
```
[the remark this answers is not in the document]
Comment (avalmas-programize) [reply packages/api/scripts/backfill-sofar-time-series.ts:72]: Done.

File: packages/api/scripts/backfill-sofar-time-series.ts [hunk 7]
```diff
@@ -32,7 +66,13 @@ async function run() {
...
   const timeSeriesRepository = connection.getRepository(TimeSeries);
 
-  addSpotterData(parsedReefIds, days, connection, {
+  const fn = getTaskFn(task);
+
+  if (!fn) {
+    throw new Error(`Task ${task} does not exist`);
+  }
+
+  return fn(parsedReefIds, days, connection, {
```
Comment (ericboucher) [inline packages/api/scripts/backfill-sofar-time-series.ts:71]: don't we need an await somewhere?

**Review by ericboucher — APPROVED** [review]

**Pull request merged**